src/UI/Implementation/Component/Chart/Speedo/Renderer.php

```diff
@@ -0,0 +1,295 @@
+<?php
+
+/* Copyright (c) 2017 Ralph Dittrich <dittrich@qualitus.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Chart\Speedo;
+
+use ILIAS\UI\Implementation\Render\AbstractComponentRenderer;
+use ILIAS\UI\Renderer as RendererInterface;
+use ILIAS\UI\Component;
+
+/**
+ * Class Renderer
+ * @package ILIAS\UI\Implementation\Component\Chart\Speedo
+ */
+class Renderer extends AbstractComponentRenderer
+{
+
+    /**
+     * @inheritdocs
+     */
+    public function render(Component\Component $component, RendererInterface $default_renderer)
+    {
+        /**
+         * @var Component\Chart\Speedo\Speedo $component
+         */
+        $this->checkComponent($component);
+
+        if ($component instanceof Component\Chart\Speedo\Responsive) {
+            /**
+             * @var Component\Chart\Speedo\Responsive $component
+             */
+            return $this->renderResponsive($component, $default_renderer);
+
+        } elseif ($component instanceof Component\Chart\Speedo\Mini) {
+            /**
+             * @var Component\Chart\Speedo\Mini $component
+             */
+            return $this->renderMini($component, $default_renderer);
+
+        } else {
+            /**
+             * @var Component\Chart\Speedo\Standard $component
+             */
+            return $this->renderStandard($component, $default_renderer);
+        }
+
+    }
+
+    /**
+     * Render standard speedo
+     *
+     * @param Component\Component $component
+     * @param RendererInterface $default_renderer
+     * @return string
+     */
+    protected function renderStandard(Component\Chart\Speedo\Standard $component, RendererInterface $default_renderer)
+    {
+        if ($component->hasDiagnostic()) {
+            $tpl = $this->getTemplate("tpl.speedo_two_bar.html", true, true);
+        } else {
+            $tpl = $this->getTemplate("tpl.speedo_one_bar.html", true, true);
+        }
+
+        // set "responsive class" false
+        $tpl->setVariable("RESP_CLASS", 'standard');

```

Comment:
Do not set class names through php, this makes templates unreadable for designers. Set them by setting up a section that you parse here. See e.g. the highlighted class in the glyph.html:
`<!-- BEGIN highlighted --> highlighted<!-- END highlighted -->`

Reply:
Thank you, this is a good advice! 👍 I did not know about this option till now.